pkg/provider/cloud/kubevirt/provider.go

```diff
@@ -177,3 +184,22 @@ func GetCredentialsForCluster(cloud kubermaticv1.CloudSpec, secretKeySelector pr
 
 	return kubeconfig, nil
 }
+
+func (k *kubevirt) updateInfraStorageClassesInfo(ctx context.Context, spec *kubermaticv1.CloudSpec) error {
+	kubeConfig, err := GetCredentialsForCluster(*spec, k.secretKeySelector)
+	if err != nil {
+		return err
+	}
+	client, _, err := NewClientWithRestConfig(kubeConfig)
```

Comment:
There is no need to create your own client instance here, we have a function to list storage class already
https://github.com/kubermatic/kubermatic/blob/master/pkg/handler/common/provider/kubevirt.go#L207

Reply:
as discussed with @sankalp-r, that was a bad idea, @sankalp-r will do small refactoring and move the functions to the provider that we pull provider pkg -> handler common.